JAMES-2586 Add search module chooser for Postgres app

Arsnael opened the pull request.

chibenwa commented on server/apps/postgres-app/pom.xml:
```diff
@@ -151,6 +179,12 @@
         </dependency>
+        <dependency>
+            <groupId>${james.groupId}</groupId>
+            <artifactId>james-server-jmap-draft-integration-testing</artifactId>
```
Question: what is this needed for?

chibenwa commented on server/container/guice/mailbox-postgres/src/main/java/org/apache/james/modules/mailbox/LuceneSearchMailboxModule.java:
```diff
@@ -1,58 +0,0 @@
-import com.google.inject.Singleton;
-import com.google.inject.multibindings.Multibinder;
-
-public class LuceneSearchMailboxModule extends AbstractModule {
```
Please remove the Lucene dependency, then!

chibenwa commented on server/container/guice/mailbox-postgres/src/main/java/org/apache/james/modules/mailbox/PostgresMailboxModule.java:
```diff
@@ -85,7 +85,6 @@ protected void configure() {
         postgresDataDefinitions.addBinding().toInstance(PostgresMailboxAggregateModule.MODULE);
 
         install(new PostgresQuotaModule());
-        install(new PostgresQuotaSearchModule());
```
Could you confirm quota search is well implemented with OpenSearch?

Arsnael replied:
It is cause when I had this, test were failing on a double injection?

Arsnael replied:
To be a bit more precise they are being injected by the module chooser

chibenwa reviewed (commented):
Nice work.

I want it to be easy to start this. Please add:

 - Folder `sample-configuration-distributed` with sample file `opensearch.properties`
 - Sample docker compose `docker-compose-distributed.yml`
 - A line in the README on how to start that  `docker-compose-distributed.yml`

vttranlina commented on server/apps/postgres-app/src/test/java/org/apache/james/PostgresWithTikaTest.java:
```diff
@@ -0,0 +1,44 @@
+            .build())
+        .server(PostgresJamesServerMain::createServer)
+        .extension(new DockerOpenSearchExtension())
+        .extension(new TikaExtension())
```
Q: Will we support tika in Postgres?

(just re-confirm)

Arsnael replied:
Some injections with the search module chooser required the tika module. So I had to test it. But if not tika properties file, it is disabled and we use jsoupextractor instead. From what I understand when checking quickly

Arsnael commented:
Added the docker compose distributed with openSearch.

Also while I was at it, reworked the readme to remove all JPA mentions, propose use of postgres app with only postgres or distributed (opensearch), rework the normal docker compose to have scanning conf instead of opensearch

chibenwa commented on server/apps/postgres-app/README.adoc:
```diff
@@ -26,15 +59,27 @@ Once everything is set up, you just have to run the jar with:
+
+Then you need to generate the keystore, rebuild the application jar and run it like above.
+
 == Docker distribution
```
IMO we can remove this docker section and just replace it by docker-compose.

chibenwa commented on server/apps/postgres-app/sample-configuration-single/search.properties:
```diff
@@ -0,0 +1 @@
+implementation=scanning
```
Add a comment: `not for production purposes. To be replaced by PG based search.`

Arsnael commented:
Squashed and rebased
